QuizApp/NetworkService.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  NetworkService.swift
+//  QuizApp
+//
+//  Created by Marin on 10.05.2021..
+//
+
+import UIKit
+
+class NetworkService: NetworkServiceProtocol {
+    private let baseURL: String! = "iosquiz.herokuapp.com"
+    private let apiToken: String! = "api_token"
+    private let userId: String! = "user_id"
+    private let userDefaults = UserDefaults.standard
+    
+    func login(username: String, password: String, completionHandler: @escaping (Result<Bool, RequestError>) -> Void) {
+        let session = URLSession(configuration: URLSessionConfiguration.default)
```

Comment:
It is preferred to have one URL session for many tasks, instead of creating a new session for each task.